Script to set the connector name

hariso opened the pull request:
### Description

Usage: `./setup.sh mysql` (for example).

### Quick checks:

- [ ] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [ ] There is no other [pull request](https://github.com/ahamidi/conduit-connector-template/pulls) for the same update/change.
- [ ] I have written unit tests.
- [ ] I have made sure that the PR is of reasonable size and can be easily reviewed.

hariso commented on .github/pull_request_template.md:
```diff
@@ -9,6 +9,6 @@ Fixes # (issue)
 ### Quick checks:
 
 - [ ] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
```
@ahamidi @lovromazgon WDYT about removing this?

lovromazgon replied:
I lean towards removing it. For our own connectors, it might make sense to have this check, but even if we don't it's not the end of the world. For 3rd party connectors it does not make sense to have this check, they are free to choose their own guidelines. And given that we're creating a public template I think we should not include the check.

ahamidi replied:
Yep, let's drop it.

hariso replied:
Dropped!

hariso commented on destination_test.go:
```diff
@@ -1,39 +1,26 @@
 	}
 }
 
-func TestConfigureDestination_FailsWhenConfigInvalid(t *testing.T) {
-	con := connector.Destination{}
-	err := con.Configure(context.Background(), map[string]string{"foobar": "foobar"})
-	if err != nil {
-		t.Errorf("expected no error, got %v", err)
-
-	}
-
-	if strings.HasPrefix(err.Error(), "config is missing:") {
-		t.Errorf("expected error to be about invalid config, got %v", err)
-	}
-}
```
I felt that the code to make this pass would be too "opinionated", since it can be done in multiple ways.

lovromazgon commented on .github/CODEOWNERS:
```diff
@@ -1 +0,0 @@
-*       @ahamidi
```
I'd add a comment here that describes what this file does, maybe with a link to github docs about codeowners.

hariso replied:
Yup, that makes sense. :+1:

lovromazgon commented on .github/workflows/dependabot-auto-merge-go.yml:
```diff
@@ -18,7 +18,7 @@ jobs:
     steps:
```
Should we also add the [release action](https://github.com/ConduitIO/conduit-connector-file/blob/main/.github/workflows/release.yml)?

hariso replied:
I think that's a good idea.

lovromazgon commented on test/docker-compose-template.yml:
```diff
@@ -0,0 +1,4 @@
+version: "3.9"
```
I'd rename this to `docker-compose.yml` and add a comment on top describing what it's for and how it can be adjusted.

hariso replied:
Makes sense. :+1:

lovromazgon commented on setup.sh:
```diff
@@ -0,0 +1,25 @@
+  find . -type f ! -name "setup.sh" -exec sed -i "s~connectorName~$CONNECTOR_NAME~g" {} +
+fi
+
+# Remove this script
```
Should we also remove `README.md` and `mv README_TEMPLATE.md README.md`?

hariso replied:
Good suggestion. :+1:

lovromazgon commented on README.md:
```diff
@@ -1,8 +1,31 @@
 ### Conduit Connector Template
```
⛏️ 
```suggestion
# Conduit Connector Template
```

lovromazgon commented on README.md:
```diff
@@ -1,8 +1,31 @@
+This template includes the following:
+* Skeleton code for the connector's configuration, source and destination
+* Example unit tests
+* A [Makefile](/Makefile) which commonly used targets
```
```suggestion
* A [Makefile](/Makefile) with commonly used targets
```

lovromazgon commented on README.md:
```diff
@@ -1,8 +1,31 @@
 
 ### How to use
-Clone this repo and implement the various relevant methods.
+* On this repository's main page, click the "Use this template"
```
```suggestion
* On this repository's main page, click "Use this template"
```

lovromazgon commented on README_TEMPLATE.md:
```diff
@@ -1,34 +1,36 @@
-### Conduit Connector for <resource>
+## Conduit Connector for <resource>
```
Would this make sense? Then the name would also be replaced by the script. If not, I'd at least change the header to be the top most header.
```suggestion
# Conduit Connector for connectorName
```

hariso replied:
The reason I left it as <resource> is that it should a "prettier" name, as it would appear in a normal sentence (e.g. MySQL, Amazon S3 etc.).

lovromazgon replied:
Makes sense, I'd still change the header level though.
```suggestion
# Conduit Connector for <resource>
```

lovromazgon commented on .github/CODEOWNERS:
```diff
@@ -1 +1,3 @@
-*       @ahamidi
+# Define code owners (individuals or teams that are responsible for code in this repository)
+# More about code owners at https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners
+*       @ConduitIO/conduit-core
```
Shouldn't this be a comment?
```suggestion
# Example:
# * @ConduitIO/conduit-core
```

hariso replied:
Once this is moved to a repo under conduitio, conduit-core should be automatically asked for a review, which is I added this. It will be removed by the setup script, and I believe the link to the docs about code owners is also sufficient (i.e. an example is not a must have in this case).

lovromazgon commented on .github/workflows/build.yml:
```diff
@@ -14,7 +14,7 @@ jobs:
         uses: actions/setup-go@v3
         with:
-          go-version: 1.18
+          go-version: 1.19
```
Since we're using go 1.19 in the actions we should also change the version in `go.mod` to 1.19.

hariso replied:
Already done. :+1:

lovromazgon replied:
Oh my bad, I must have been looking at the wrong repo.

lovromazgon reviewed (commented):
Last round of comments, looks good otherwise 👍

lovromazgon reviewed (approved):
🚀

lovromazgon commented:
I just realized the last section in the readme is outdated (Configuration), maybe it's not even needed. Also the section Specification is mentioning configuration options, that's not true anymore, it only contains some general metadata about the connector.

hariso commented:
> I just realized the last section in the readme is outdated (Configuration), maybe it's not even needed. Also the section Specification is mentioning configuration options, that's not true anymore, it only contains some general metadata about the connector.

Good catch! I've updated the spec part. Regarding the configuration: IMHO it's useful and it's worth having it. I've updated it so that it reflects the current, flat, structure.
